webapp/src/components/llmbot_post/utils.ts

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2023-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {PermalinkData} from './types';
+
+export function extractPermalinkData(post: any): PermalinkData | null {
```

Comment:
This might be better organized as a file called `permalink_data.ts` with the type and this utility in it.